sdk/schemaregistry/Microsoft.Azure.Data.SchemaRegistry.ApacheAvro/src/SchemaRegistryAvroEncoder.cs

```diff
@@ -58,19 +54,114 @@ private enum SupportedType
             GenericRecord
         }
 
-        private static SupportedType GetSupportedTypeOrThrow(Type type)
+        #region Encode
+
+        /// <summary>
+        /// Encodes the message data into Avro and stores it in <see cref="MessageWithMetadata.Data"/>. The <see cref="MessageWithMetadata.ContentType"/>
+        /// will be set to "avro/binary+schemaId" where schemaId is the ID of the schema used to encode the data.
+        /// </summary>
+        /// <param name="data">The data to serialize to Avro and encode into the message.</param>
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
+        /// <typeparam name="TMessage">The <see cref="MessageWithMetadata"/> type to encode the data into.</typeparam>
+        /// <typeparam name="TData">The type of the data to encode.</typeparam>
+        public TMessage EncodeMessageData<TMessage, TData>(
+            TData data,
+            CancellationToken cancellationToken = default) where TMessage : MessageWithMetadata, new()
+            => (TMessage) EncodeMessageDataInternalAsync(data, typeof(TData), typeof(TMessage), false, cancellationToken).EnsureCompleted();
+
+        /// <summary>
+        /// Encodes the message data into Avro and stores it in <see cref="MessageWithMetadata.Data"/>. The <see cref="MessageWithMetadata.ContentType"/>
+        /// will be set to "avro/binary+schemaId" where schemaId is the ID of the schema used to encode the data.
```

Comment:
Should we consider making it more clear at a glance that "schemaId" isn't a literal by decorating it with brackets or braces or something?   

```suggestion
        /// will be set to "avro/binary+<<schemaId>>" where schemaId is the ID of the schema used to encode the data.
```